src/lib/auxiliary_database/external_transition_database.ml

```diff
@@ -1,14 +1,128 @@
 open Core
 open Coda_base
 open Coda_transition
+open Signature_lib
 module Time = Block_time
 
+(** Block_data is the external_transition data that GraphQL needs *)
+module Block_data = struct
+  module Transactions = struct
+    module Stable = struct
+      module V1 = struct
+        module T = struct
+          type t =
+            { user_commands: User_command.Stable.V1.t list
+            ; fee_transfers: Fee_transfer.Single.Stable.V1.t list
+            ; coinbase: Currency.Amount.Stable.V1.t }
+          [@@deriving bin_io, version {unnumbered}]
+        end
+
+        include T
+      end
+
+      module Latest = V1
+    end
+
+    type t = Stable.Latest.t =
+      { user_commands: User_command.t list
+      ; fee_transfers: Fee_transfer.Single.t list
+      ; coinbase: Currency.Amount.t }
+  end
+
+  module Protocol_state = struct
+    module Stable = struct
+      module V1 = struct
+        module T = struct
+          type t =
+            { previous_state_hash: State_hash.Stable.V1.t
+            ; blockchain_state: Coda_state.Blockchain_state.Value.Stable.V1.t
+            }
+          [@@deriving bin_io, version {unnumbered}]
+        end
+
+        include T
+      end
+
+      module Latest = V1
+    end
+
+    type t = Stable.Latest.t =
+      { previous_state_hash: State_hash.t
+      ; blockchain_state: Coda_state.Blockchain_state.Value.t }
+  end
+
+  module Stable = struct
+    module V1 = struct
+      module T = struct
+        type t =
+          { creator: Public_key.Compressed.Stable.V1.t
+          ; protocol_state: Protocol_state.Stable.V1.t
+          ; transactions: Transactions.Stable.V1.t }
+        [@@deriving bin_io, version {unnumbered}]
+      end
+
+      include T
+    end
+
+    module Latest = V1
+  end
+
+  type t = Stable.Latest.t =
+    { creator: Public_key.Compressed.t
+    ; protocol_state: Protocol_state.t
+    ; transactions: Transactions.t }
+
+  let of_transition external_transition =
```

Comment:
I think here you'd also want to take in `~tracked_participants` or something, a set of participants you care about tracking. And then you only need to store the transactions involving the participants (you can call `~tracked_participants:(Wallets.pks t.wallets |> Set.of_list)` or something)

Reply:
What happens if we start tracking a new participant thought?  A block that we collected could contain transactions involving the new participant. We could have filtered out the new participant's transactions, which could lead to a bad UI experience.

Reply:
Good point, but I think we have to make this sacrifice to deliver on the “succinct blockchain promise”, but cc @o1pranay to break the tie.

Reply:
seems like this is a bigger discussion we need to have about how to support adding a new participant (eg. importing a wallet).  We definitely do not want to keep all the blocks just so later we can grab txns for an arbitrary public key - defeats the purpose of a succinct blockchain as @bkase mentioned.  But we def do want to allow importing a wallet, so maybe we should discuss using transaction receipts to get past txn data - I'm not too sure how this would work, so would need some help understanding how we would do this.

For now though, since we won't support importing a wallet either way for mvp release, we can punt on this - I defer to y'all on which would allow more extensibility for when we do support it

Reply:
Okay, for extensibility, I will not implement @bkase's suggestion for this PR.